### DISCUSSION TITLE
Is this library support building ros node?

### DISCUSSION THREAD

**Discussion by Hubuwenxin** [Rubberazer/JETGPIO#28] [Q&A] [answered]
When i write ros node to publish the data which is gathered by SPI, here is the error:
 "nvidia@nvidia-desktop:~/catkin_ws$  rosrun spi_loop_pkg talker
/dev/mem: Operation not permitted
Please run this program as root (for example with sudo)
Jetgpio initialisation failed. Error code: -1".
It's sound like that can't access the GPIO library.

In the Cmakelist.txt, i add : 
include_directories(
  ${catkin_INCLUDE_DIRS}
  /home/nvidia/sound/JETGPIO)
target_link_libraries(talker 
${catkin_LIBRARIES} 
/usr/lib/libjetgpio.so)

I'd like to inquiry that Is this library support building ros node?I'll appreciate that If you can reply me. Thank you so much!

**Comment by Rubberazer** [comment, marked as answer]
You should use sudo, like: 

`sudo rosrun spi_loop_pkg talker`

**Reply by ambulantelab** [reply]
Is there another option for this? Having to run the nodes as sudo is cumbersome especially since all other nodes have to be sudo as well to communicate?

**Reply by Rubberazer** [reply]
I suggest you read this: https://github.com/Rubberazer/JETGPIO/issues/32
Unfortunately I have never had ROS integration in mind when I created this library, option 3 on my original answer on the issue linked above works, but is rather cumbersome...
Sorry but no plans to change this at this point. No time, neither enough incentives to do so.
